README.md

```diff
@@ -69,33 +70,161 @@ ckan-client-js
+//   id: '03de2e7a-6e52-4410-b6b1-49491f0f4d5a',
+//   metadata_created: '2020-09-16T15:03:18.022114',
+//   metadata_modified: '2020-09-16T15:07:51.299795',
+//   creator_user_id: 'cdb427df-c1ac-4365-b33c-94ccfad55aff',
+//   type: 'dataset',
+//   resources: [
+//     {
+//       hash: '',
+//       description: '',
+//       format: 'CSV',
+//       path: 'https://somecsvonline.com/somecsv.csv',
+//       package_id: '03de2e7a-6e52-4410-b6b1-49491f0f4d5a',
+//       created: '2020-09-16T15:07:51.315447',
+//       revision_id: '60460e2f-c22b-4107-bee2-ccb21c849054',
+//       id: '9f01b4a5-9592-4528-b135-5c0ddd43720c',
+//       bytes: 12
+//     }
+//   ],
+//   groups: [],
+//   relationships_as_subject: [],
+//   name: 'market',
+//   title: 'market',
+//   revision_id: 'cfae5ff5-9b2e-4c91-965d-c0a2c740da37'
+// }
+```
 
-const client = new Client('key', 'organization-name', 'dataset-name', 'api')
-const resource = f11s.open(file)
+If you want to make more advanced requests to CKAN API, then you can use `action()` method. Please note that it accept CKAN dataset and returns CKAN dataset. If you want to have frictionless data you have to use [CKAN<=>Frictionless Mapper](https://github.com/datopian/frictionless-ckan-mapper-js)
 
-client.pushBlob(resource, onUploadProgress)
+```js
+// Update the the dataset name
+const response = await client.action('package_update', {
+  id: '03de2e7a-6e52-4410-b6b1-49491f0f4d5a',
+  name: 'market1',
+})
+console.log(response.result)
+// {
+//   license_title: null,
+//   maintainer: null,
+//   relationships_as_object: [],
+//   private: false,
+//   maintainer_email: null,
+//   num_tags: 0,
+//   id: '03de2e7a-6e52-4410-b6b1-49491f0f4d5a',
+//   metadata_created: '2020-09-16T15:03:18.022114',
+//   metadata_modified: '2020-09-16T15:16:17.696326',
+//   author: null,
+//   author_email: null,
+//   state: 'active',
+//   version: null,
+//   creator_user_id: 'cdb427df-c1ac-4365-b33c-94ccfad55aff',
+//   type: 'dataset',
+//   resources: [],
+//   num_resources: 0,
+//   tags: [],
+//   groups: [],
+//   license_id: null,
+//   relationships_as_subject: [],
+//   organization: null,
+//   name: 'market1',
+//   isopen: false,
+//   url: null,
+//   notes: null,
+//   owner_org: null,
+//   extras: [],
+//   title: 'market',
+//   revision_id: '2b3fc86b-fcc0-47cc-92f2-a6c4830638f4'
+// }
+```
 
-const onUploadProgress = progressEvent => {
-  let progress = (progressEvent.loaded / progressEvent.total) * 100
-  console.log(progress)
+Uploading file and updating the dataset
+
+```js
+// This file can be also a browser File attached by the user, if it's in browser environment
+const file = {
+  name: 'mydata',
+  data: { foo: 'bar' },
 }
+
+const resource = f11s.open(file)
+// If it is in Node you can also get the resource by reading the file
+// const resource = f11s.open('path/to/file')
+
+client.pushBlob(resource)
+// If you are in browser you can also track the progress, in the second argument
```

Comment:
No unfortunately https://github.com/axios/axios/issues/1966 The lib is using axios, this limitation comes from axios

Reply:
Are you sure? In https://github.com/datopian/datahub-client IIRC we used axios and do progress reporting on upload. Can you open an issue in this repo to investigate this (at some point in the future).

Reply:
@kmanaseryan ^^^

Reply:
@rufuspollock I haven't tried it myself but what the community says it can't be used in Node since it `onUploadProgress` is based on xhr. Read the comment here
https://github.com/axios/axios/issues/1966#issuecomment-464963763

Will open the issue now

Reply:
Doen: https://github.com/datopian/datahub-client/issues/44